R/qenv-eval_code.R

```diff
@@ -32,6 +32,8 @@ setMethod("eval_code", signature = c("qenv", "expression"), function(object, cod
   for (code_line in code) {
     # Using withCallingHandlers to capture ALL warnings and messages.
     # Using tryCatch to capture the FIRST error and abort further evaluation.
+
+    grDevices::dev.new()
```

Comment:
I think now that we could do it like this - to disable graphical device only in shiny session or cicd. I also think whether moving this to tryCatch is a better idea
```suggestion
    if (!interactive()) {
      grDevices::dev.new()
      on.exit(grDevices::dev.off())
    }
```

Reply:
Yes, we can definitely move this into a tryCatch block. Initially, I did not include it within the tryCatch because it wasn't closing the new graphic device in case of an error or warning. Additionally, this should be` if (interactive()) {...}` to ensure it runs only in interactive mode.